pkg/openfeature/client_test.go

```diff
@@ -0,0 +1,149 @@
+		GetStringValueDetails(flag string, defaultValue string, evalCtx EvaluationContext, options ...EvaluationOption) (EvaluationDetails, error)
+		GetNumberValueDetails(flag string, defaultValue float64, evalCtx EvaluationContext, options ...EvaluationOption) (EvaluationDetails, error)
+		GetObjectValueDetails(flag string, defaultValue interface{}, evalCtx EvaluationContext, options ...EvaluationOption) (EvaluationDetails, error)
+	}
+
+	var clientI interface{} = client
+	if _, ok := clientI.(requirements); !ok {
+		t.Error("client returned by GetClient doesn't implement the 1.4.1 requirements interface")
+	}
+}
+
+// Requirement_1_4_2 has no suitable test as the provider implementation populates the EvaluationDetails value field
+
+// TODO Requirement_1_4_3 once upgraded Go to 1.18 for generics
+
+func TestRequirement_1_4_4(t *testing.T) {
+	defer t.Cleanup(initSingleton)
+	client := GetClient("test-client")
+
+	flagKey := "foo"
+
+	t.Run("GetBooleanValueDetails", func(t *testing.T) {
+		evDetails, err := client.GetBooleanValueDetails(flagKey, true, nil)
+		if err != nil {
+			t.Error(err)
+		}
+		if evDetails.FlagKey != flagKey {
+			t.Errorf(
+				"flag key isn't as expected in EvaluationDetail, got %s, expected %s",
+				evDetails.FlagKey, flagKey,
+			)
+		}
+	})
+
+	t.Run("GetStringValueDetails", func(t *testing.T) {
+		evDetails, err := client.GetStringValueDetails(flagKey, "", nil)
+		if err != nil {
+			t.Error(err)
+		}
+		if evDetails.FlagKey != flagKey {
+			t.Errorf(
+				"flag key isn't as expected in EvaluationDetail, got %s, expected %s",
+				evDetails.FlagKey, flagKey,
+			)
+		}
+	})
+
+	t.Run("GetNumberValueDetails", func(t *testing.T) {
+		evDetails, err := client.GetNumberValueDetails(flagKey, 1, nil)
+		if err != nil {
+			t.Error(err)
+		}
+		if evDetails.FlagKey != flagKey {
+			t.Errorf(
+				"flag key isn't as expected in EvaluationDetail, got %s, expected %s",
+				evDetails.FlagKey, flagKey,
+			)
+		}
+	})
+
+	t.Run("GetObjectValueDetails", func(t *testing.T) {
+		evDetails, err := client.GetObjectValueDetails(flagKey, 1, nil)
+		if err != nil {
+			t.Error(err)
+		}
+		if evDetails.FlagKey != flagKey {
+			t.Errorf(
+				"flag key isn't as expected in EvaluationDetail, got %s, expected %s",
+				evDetails.FlagKey, flagKey,
+			)
+		}
+	})
+}
+
+// Requirement_1_4_5 has no suitable test as the provider implementation populates the EvaluationDetails variant field
+
+// Requirement_1_4_6 has no suitable test as the provider implementation populates the EvaluationDetails reason field
+
+// Requirement_1_4_7 has no suitable test as the provider implementation populates the EvaluationDetails error code field
+
+// Requirement_1_4_8 has no suitable test as the provider implementation populates the EvaluationDetails reason field
+
+// Requirement_1_4_9 is satisfied by the error included in the return signatures, as is idiomatic in Go. Errors aren't
+// fatal, the operations won't terminate as a result of an error.
```

Comment:
It MAY be worth testing that in the case of abnormal execution of the provider, the default is returned as well as an error, but I don't feel too strongly about it.

Reply:
Done